docs/proposals/caching/README.md

```diff
@@ -0,0 +1,254 @@
+---
+title: Caching member cluster resources for Karmada
+authors:
+- "@huntsman-li"
+- "@liys87x"
+reviewers:
+approvers:
+
+creation-date: 2022-02-14
+
+---
+
+# Caching member cluster resources for Karmada
+
+## Summary
+
+Provides a caching layer for Karmada to cache (member clusters) Kubernetes resources.
+
+## Motivation
+
+### Goals
+
+- Accelerates resource requests processing speed across regions
+- Provides a cross-cluster resource view
+- Compatible with (multi-cluster) multiple kubernetes resource versions
+- Unified resource requests entries
+- Reduces API server pressure of member clusters
+
+### Non-Goals
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+Imagine that we have a cross-region application that needs to run in multiple regions in order to provide service capabilities nearby. We achieve this by deploying a Kubernetes cluster in each region.
+
+Goals:
+
+- Get the distribution of the application in different clusters.
+- Get resource information across clusters through a unified endpoint.
+- Accelerate processing speed of resource requests across regions.
+- Get resources in multiple clusters by labels.
+
+## Design Details
+
+### Define the scope of the cached resource
+
+#### New ClusterCache API
+
+We propose a new CR in `cluster.karmada.io` group.
```

Comment:
Let's postpone making decisions for the group, as maybe we can introduce a new group, like `search.karmada.io`.